compiler/src/iree/compiler/Codegen/LLVMGPU/Utils/LLVMGPULayoutAnalysisAndDistribution.cpp

```diff
@@ -693,41 +715,185 @@ void distributeReductionBroadcastTranspose(
   else
     simdToSimtMap.try_emplace(broadcastOp.getResult(), output);
 
-  ops.push_back(reductionOp);
-  ops.push_back(broadcastOp);
-  if (transposeOp) ops.push_back(transposeOp);
+  ops.insert(reductionOp);
+  ops.insert(broadcastOp);
+  if (transposeOp) ops.insert(transposeOp);
+}
+
+static void replaceForOpWithNewSignature(
+    RewriterBase &rewriter, scf::ForOp loop, ValueRange newIterOperands,
+    DenseMap<Value, Layout> &layoutMap, DenseMap<Value, Value> &simdToSimtMap) {
+  OpBuilder::InsertionGuard g(rewriter);
+  rewriter.setInsertionPoint(loop);
+
+  // Create a new loop before the existing one, with the extra operands.
+  // We will be using dummy values instead of the old operands
+  // only for those operands that are being distributed
+  SmallVector<Value> dummyOperands, newOperands;
+  auto operands = llvm::to_vector<4>(loop.getIterOperands());
+  for (auto operand : operands) {
+    if (!layoutMap.count(operand)) {
+      dummyOperands.push_back(operand);
+      newOperands.push_back(operand);
```

Comment:
When there is no layout, we just leave the operand as is. Otherwise we push a zero constant. I will remove dummy operands as it is not being used anywhere else.